types/wlr_scene.c

```diff
@@ -226,6 +276,27 @@ static void scissor_output(struct wlr_output *output, pixman_box32_t *rect) {
 	wlr_renderer_scissor(renderer, &box);
 }
 
+static void render_rect(struct wlr_output *output,
+		pixman_region32_t *output_damage, const float color[static 4],
+		const struct wlr_box *box, const float matrix[static 9]) {
+	struct wlr_renderer *renderer = wlr_backend_get_renderer(output->backend);
+	assert(renderer);
+
+	pixman_region32_t damage;
+	pixman_region32_init(&damage);
+	pixman_region32_init_rect(&damage, box->x, box->y, box->width, box->height);
+	pixman_region32_intersect(&damage, &damage, output_damage);
+	if (pixman_region32_not_empty(&damage)) {
+		int nrects;
+		pixman_box32_t *rects = pixman_region32_rectangles(&damage, &nrects);
+		for (int i = 0; i < nrects; ++i) {
+			scissor_output(output, &rects[i]);
+			wlr_render_rect(renderer, box, color, matrix);
```

Comment:
Technically we can just use clear() here

Reply:
If we do that, we should be able to remove the `matrix` arg.

Reply:
clear() as in `wlr_renderer_clear()`? Won't that cause issues with semi-transparent RECT nodes?

Reply:
Derp. Indeed, we need to always use`wlr_render_rect`.